src/js/components/Popup/PopupConfirm.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from "react";
+
+interface State {
+  isPopupOpen: boolean;
+}
+
+interface Props {
+  buttonText: string | React.ReactElement;
+  popUpText: string | React.ReactElement;
+  popupStyle?: React.CSSProperties;
+  direction?: "down" | "right" | "left" | "rightHang" | "leftHang" | "overlay";
+  onAccept: () => void;
+  onCancel?: () => void;
+}
+
+export default class Popup extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      isPopupOpen: false
+    };
+  }
+
+  componentDidMount() {
+    document.body.addEventListener("click", this.handleBodyClick.bind(this));
+  }
+
+  componentWillUnmount() {
+    document.body.removeEventListener("click", this.handleBodyClick.bind(this));
+  }
+
+  handleBodyClick() {
+    this.setState({
+      isPopupOpen: false
+    });
+  }
+
+  togglePopup() {
+    let isPopupOpen = this.state.isPopupOpen;
+    this.setState({
+      isPopupOpen: !isPopupOpen
+    });
+  }
+
+  render() {
+    return (
+      <div className="Popup">
+        <button
+          className="Button Button--small"
+          onClick={this.togglePopup.bind(this)}
+        >
+          {this.props.buttonText}
+        </button>
+        <div
+          className={
+            "Popup-container " +
```

Comment:
I like this approach for generic popup. It might make sense to drop the confirm part of the naming, since it could be used for any number of popups, like player cards in the future. Confirms would always be overlays. And again can will this able to take multiple arguments (Up, Left), (Down, Right), etc?

Reply:
Actually with this one since it's just a generic cancel + accept with text, maybe just go with the overlay without options. A more generic popup that accepted any content and options for style and sizing would be cool though

Reply:
Dropped the options and have it set to an overlay. We may want to revisit this at some point for a generic (whatever you want goes inside) pop over.